### PR TITLE
FILTER: ExtractInternalSurfacesFromTriangleGeometry

### PR DESCRIPTION
**Pull request by mmarineBlueQuartz**
* Added ExtractInternalSurvacesFromTriangleGeom filter
* Added unit test
* Added CreateTriangleGeomAction
* Renamed DataObject::getParentDataPaths to getDataPaths due to misleading name. DataObject::getParentDataPaths did not return DataPaths to known parents. It returned the DataPaths to the object calling it.

### CODE REVIEW

**Review by JDuffeyBQ — COMMENTED** [review]
The `DataObject::getParentDataPaths()` changes should probably be in a separate PR. And I thought there was already a change about that function?

**Comment by imikejackson** [comment]
@mmarineBlueQuartz  What is the function `getDataPaths()` *supposed* to return? Because it returns the DataPaths of all the parents of the object. If it is *supposed* to return the children DataPaths then there is a logic error in the function.

File: src/Plugins/ComplexCore/src/ComplexCore/Filters/ExtractInternalSurfacesFromTriangleGeometry.cpp [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+    {
+      usize hash = std::hash<float32>()(vert[0]);
+      hashCombine(hash, vert[1]);
+      hashCombine(hash, vert[2]);
+      return hash;
+    }
+  };
+
+  typedef std::array<int64, 3> Triangle;
+  typedef std::unordered_map<Vertex, int64, ArrayHasher> VertexMap;
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/src/ComplexCore/Filters/ExtractInternalSurfacesFromTriangleGeometry.cpp:224]:
```suggestion
  using Triangle = std::array<int64, 3>;
  using VertexMap = std::unordered_map<Vertex, int64, ArrayHasher>;
```

**Review by JDuffeyBQ — CHANGES_REQUESTED** [review]
This just needs the `getDataPaths()` changes extracted to a separate PR with whatever name we ended up deciding on.

**Review by imikejackson — APPROVED** [review]

**Pull request merged**